src/Microsoft.DocAsCode.MarkdigEngine.Extensions/Row/RowParser.cs

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.DocAsCode.MarkdigEngine.Extensions
+{
+    using Markdig.Helpers;
+    using Markdig.Parsers;
+    using Markdig.Syntax;
+    using Microsoft.DocAsCode.Common;
+
+    public class RowParser : BlockParser
+    {
+        private const string StartString = "row";
+        private const string EndString = "row-end:::";
+        private const char Colon = ':';
+        public RowParser()
+        {
+            OpeningCharacters = new[] { ':' };
+        }
+
+        public override BlockState TryOpen(BlockProcessor processor)
+        {
+            if (processor.IsCodeIndent)
+            {
+                return BlockState.None;
+            }
+
+            var slice = processor.Line;
+            if (ExtensionsHelper.IsEscaped(slice))
+            {
+                return BlockState.None;
+            }
+
+            var column = processor.Column;
+            var sourcePosition = processor.Start;
+            var colonCount = 0;
+
+            var c = slice.CurrentChar;
+            while (c == Colon)
+            {
+                c = slice.NextChar();
+                colonCount++;
+            }
+
+            if (colonCount < 3) return BlockState.None;
+
+            SkipSpaces(ref slice);
+
+            if (!ExtensionsHelper.MatchStart(ref slice, "row:::", false))
+            {
+                return BlockState.None;
+            }
+
+            processor.NewBlocks.Push(new RowBlock(this)
+            {
+                ColonCount = colonCount,
+                Column = column,
+                Span = new SourceSpan(sourcePosition, slice.End),
+            });
+
+            return BlockState.ContinueDiscard;
+        }
+
+        public override BlockState TryContinue(BlockProcessor processor, Block block)
+        {
+            if (processor.IsBlankLine)
+            {
+                return BlockState.Continue;
+            }
+
+            var slice = processor.Line;
+            var Row = (RowBlock)block;
+
+            SkipSpaces(ref slice);
+
+            if (!ExtensionsHelper.MatchStart(ref slice, new string(':', Row.ColonCount)))
+            {
+                return BlockState.Continue;
+            }
+
+            SkipSpaces(ref slice);
+
+            if (!ExtensionsHelper.MatchStart(ref slice, "row-end:::", false))
+            {
+                return BlockState.Continue;
+            }
+
+            var c = SkipSpaces(ref slice);
+
+            if (!c.IsZero())
+            {
+                Logger.LogWarning($"Row has some invalid chars in the ending.");
+            }
+
+            block.UpdateSpanEnd(slice.End);
+
+            return BlockState.BreakDiscard;
+        }
+
+        public char SkipSpaces(ref StringSlice slice)
```

Comment:
>public char SkipSpaces(ref StringSlice slice) [](start = 8, length = 45)

Extract this method to ExtensionsHelper.cs since it also exists in NestedColumnParser and MonikerRangeParser?